packages/core/src/Provider/HandleProvider/types.ts

```diff
@@ -0,0 +1,23 @@
+import { ByronAddress, EnterpriseAddress, PaymentAddress, PointerAddress } from '../../Cardano/Address';
+import { Point } from '../..';
+
+export type Handle = string;
+
+type CardanoAddress = ByronAddress | PaymentAddress | EnterpriseAddress | PointerAddress;
+
+export interface HandleInfo {
+  handle: Handle;
+  hasDatum: boolean;
+  resolvedAddresses: {
+    cardano: CardanoAddress;
+  };
+  resolvedAt: Point;
+}
+
+export interface ResolveHandlesArgs {
+  handles: Handle[];
+}
+
+export interface HandleProvider {
+  resolveHandles(args: ResolveHandlesArgs): Promise<HandleInfo[] | null>;
```

Comment:
Since it can be used to resolve multiple handles, user needs to be able to determine which handle was resolved and which wasn't.
```suggestion
  resolveHandles(args: ResolveHandlesArgs): Promise<Array<HandleInfo | null>>;
```